extensions/vertx-http/deployment/src/main/java/io/quarkus/vertx/http/deployment/devmode/console/ConfigEditorProcessor.java

```diff
@@ -68,104 +45,41 @@ protected void handlePost(RoutingContext event, MultiMap form) throws Exception
+                    throw new IllegalStateException("Unable to manage configurations - no resource directory found");
                 }
-                boolean present = false;
-                if (appProps != null) {
-                    try (InputStream in = Files.newInputStream(appProps)) {
-                        appProperties.load(in);
-                        present = appProperties.containsKey(key);
-                    }
-                } else {
-                    // If there is no application.properties file then create a new one in the first module
-                    List<Path> resourcesDir = DevConsoleManager.getHotReplacementContext().getResourcesDir();
-                    if (resourcesDir.isEmpty()) {
-                        throw new IllegalStateException(
-                                "Unable to create application.properties - no resource directory found");
-                    }
-                    appProps = Files.createFile(resourcesDir.get(0).resolve("application.properties"));
+
+                // In the current project only
+                Path path = resourcesDir.get(0);
+                Path configPath = path.resolve("application.properties");
+                if (!Files.exists(configPath)) {
+                    configPath = Files.createFile(path.resolve("application.properties"));
                 }
-                if (!present) {
-                    try (OutputStream out = Files.newOutputStream(appProps, StandardOpenOption.APPEND)) {
-                        out.write(("\n" + key + "=" + value).getBytes(StandardCharsets.UTF_8)); //TODO: escpaing
-                    }
-                } else {
-                    List<String> lines = Files.readAllLines(appProps);
-                    Iterator<String> it = lines.iterator();
-                    while (it.hasNext()) {
-                        String val = it.next();
-                        if (val.startsWith(key + "=")) {
-                            it.remove();
-                        }
+
+                String profile = ProfileManager.getActiveProfile();
+                String profileKey = !profile.equals(DEVELOPMENT.getDefaultProfile()) ? "%" + profile + "." + key : key;
+                List<String> lines = Files.readAllLines(configPath);
+                for (int i = 0, linesSize = lines.size(); i < linesSize; i++) {
+                    final String line = lines.get(i);
+                    if (line.startsWith(profileKey)) {
+                        lines.set(i, profileKey + "=" + value);
+                        break;
                     }
-                    lines.add(key + "=" + value);
-                    try (BufferedWriter writer = Files.newBufferedWriter(appProps)) {
-                        for (String i : lines) {
-                            writer.write(i);
-                            writer.newLine();
-                        }
+                }
+
+                lines.add(profileKey + "=" + value);
```

Comment:
There is no guard around this any more, so it looks like if the key already exists it will be set twice?

Reply:
I guess I've removed too much code from the previous proposal :D